wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,143 @@
+import django_filters
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db.models import Count, OuterRef, Q, Subquery
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin.filters import WagtailFilterSet
+from wagtail.admin.views.reports import ReportView
+from wagtail.coreutils import get_content_languages
+from wagtail.models import ContentType, Page, PageLogEntry, get_page_models
+from wagtail.users.utils import get_deleted_user_display_name
+
+
+def _get_locale_choices():
+    choices = [
+        (language_code, display_name)
+        for language_code, display_name in get_content_languages().items()
+    ]
+    return choices
+
+
+def _annotate_last_edit_info(queryset):
+    latest_edit_log = PageLogEntry.objects.filter(content_type=OuterRef("pk")).order_by(
+        "-timestamp", "-pk"
+    )
+
+    queryset = queryset.annotate(
+        count=Count("pages"),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
+        last_edited_by=Subquery(latest_edit_log.values("user")[:1]),
+    )
+
+    return queryset
+
+
+def _annotate_last_edit_info_by_locale(queryset, language_code):
+    latest_edit_log = PageLogEntry.objects.filter(
+        content_type=OuterRef("pk"), page__locale__language_code=language_code
+    ).order_by("-timestamp", "-pk")
+
+    queryset = queryset.annotate(
+        count=Count("pages", filter=Q(pages__locale__language_code=language_code)),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
+        last_edited_by=Subquery(latest_edit_log.values("user")[:1]),
+    )
+
+    return queryset
+
+
+class LocaleFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, value):
+        if value and value != self.null_value:
+            return _annotate_last_edit_info_by_locale(qs, value)
+        else:
+            return _annotate_last_edit_info(qs)
```

Comment:
It feels like more is happening here than should be. If annotations must be made for the report to work, perhaps that should be the responsibility of `PageTypesUsageReportFilterSet`?

Reply:
This feels method relevant: https://github.com/carltongibson/django-filter/blob/main/django_filters/filterset.py#L204-L221

Reply:
I see your point, it makes total sense. Thanks for the suggestion, Andy. I've pushed some changes to address it.